catalyst_voices/packages/internal/catalyst_voices_models/lib/src/proposal/proposal.dart

```diff
@@ -123,3 +125,154 @@ final class Proposal extends Equatable {
         categoryId: categoryId ?? this.categoryId,
       );
 }
+
+final class ProposalWithVersions extends Proposal {
+  final List<ProposalVersion> versions;
+
+  factory ProposalWithVersions({
+    required DocumentRef selfRef,
+    required String title,
+    required String description,
+    required DateTime updateDate,
+    DateTime? fundedDate,
+    required Coin fundsRequested,
+    required ProposalStatus status,
+    required ProposalPublish publish,
+    required int versionCount,
+    required int duration,
+    required String author,
+    required int commentsCount,
+    required String category,
+    required List<ProposalVersion> versions,
+    required SignedDocumentRef categoryId,
+  }) {
+    final sortedVersions = List<ProposalVersion>.from(versions)..sort();
```

Comment:
`List.from` is not type safe, please use `List.of`
```suggestion
    final sortedVersions = List.of(versions)..sort();
```

Reply:
Added